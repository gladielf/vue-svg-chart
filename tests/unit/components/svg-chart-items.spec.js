import { shallowMount, mount } from '@vue/test-utils';
import SvgChartItems from '@/components/svg-chart-items';

describe('svg-chart-items - Snapshot', () => {
  it('Snapshot', () => {
    const wrapper = mount(SvgChartItems);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('svg-chart-items - Props', () => {
  it('renders props', () => {
    const wrapper = shallowMount(SvgChartItems, {
      propsData: {
        bgColor: '#fff',
        svgHeight: '1em',
        items: 1
      }
    });

    expect(wrapper.vm.bgColor).toBe('#fff');
    expect(wrapper.vm.svgHeight).toBe('1em');
    expect(wrapper.vm.items).toBe(1);
  });
});

describe('svg-chart-items - Methods', () => {
  it('onMouseOver', () => {
    const wrapper = shallowMount(SvgChartItems, {});
    expect(wrapper.vm.hovered).toBeFalsy();
    wrapper.vm.onMouseOver();
    expect(wrapper.vm.hovered).toBeTruthy();
  });
});
