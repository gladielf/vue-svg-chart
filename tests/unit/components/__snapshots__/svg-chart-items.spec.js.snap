// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svg-chart-items - Snapshot Snapshot 1`] = `
<svg
  class="chart-items"
  width="100%"
/>
`;
